### PR TITLE
feat(proxy): configure include file during checkout

### LINKED ISSUE #894: Create include file on checkout
When we checkout we should generate include file using `radicle-link` [utilities](https://github.com/radicle-dev/radicle-link/pull/300). 

So when we are [creating](https://github.com/radicle-dev/radicle-upstream/blob/master/proxy/coco/src/project/checkout.rs#L52) the working copy we should also be creating an include file for that working copy.

This means we need to place the include file in a canonical location. This means that we should use a data directory for placing the set of include files. I believe upstream already has it's own data directory, so this should be passed to the `Checkout` struct.

We should end up with something like `$XDG_DATA_HOME/includes/<project.urn.id>.inc` and the working copy should create a config entry to point to this include path.

There's an integration test called [working_copy.rs](https://github.com/radicle-dev/radicle-link/blob/master/librad/tests/working_copy.rs) to see an example use of creating the working copy with this include file.

### LINKED ISSUE #930: Keep include files fresh and clean
Once #894 has landed we should tie into significant actions affecting the tracking graph of a project to update the corresponding include file, so that local working copies can access to the most up-to-date information.

### PR DESCRIPTION
**Pull request by xla**
Closes #894 
Closes #930

### CODE REVIEW

File: proxy/coco/src/project.rs [hunk 1]
```diff
@@ -12,6 +13,18 @@ pub use create::{Create, Repo};
...
+fn set_include_path<P>(
+    repo: &git2::Repository,
+    include: &include::Include<P>,
+) -> Result<(), git2::Error>
+where
+    P: AsRef<std::path::Path>,
+{
+    let mut config = repo.config()?;
+
+    config.set_str("include.path", &include.file_path().display().to_string())
```
Comment (xla) [inline proxy/coco/src/project.rs:25]: @FintanH Can we get rid of the magic string here? Maybe `librad;:git::include` could export a const?

File: proxy/coco/src/state.rs [hunk 2]
```diff
@@ -443,6 +444,36 @@ impl State {
...
+        P: Into<Option<PeerId>>,
+    {
+        let proj = self.get_project(&urn, peer_id).map_err(Error::from)?;
+        let include = self.prepare_include(&proj).map_err(Error::from)?;
+        let checkout = project::Checkout::new(proj, destination, include);
+
+        checkout.run(self.peer_id()).map_err(Error::from)
+    }
+
+    pub fn prepare_include(
```
Comment (xla) [inline proxy/coco/src/state.rs:464]:
```suggestion
    pub fn update_include(
```

We are going to call this for many project actions as per #930

**Comment by xla** [comment]
@FintanH This is early and only does the setup during checkout in this iteration. I'm wondering what a valid strategy is to test this functionality proper.

File: proxy/coco/src/project.rs (hunk 1 above)
Comment (FintanH) [reply proxy/coco/src/project.rs]: You didn't add this after all the forrays into link? :scream:

File: proxy/coco/src/state.rs [hunk 3]
```diff
@@ -443,6 +445,48 @@ impl State {
             })
             .collect()
     }
+
+    /// Creates a workign copy for the project of the given `urn`.
```
Comment (cloudhead) [inline proxy/coco/src/state.rs:449]:
```suggestion
    /// Creates a working copy for the project of the given `urn`.
```

File: proxy/coco/src/project.rs (hunk 1 above)
Comment (xla) [reply proxy/coco/src/project.rs]: :facepalm: forgot my own todo...PR forthcoming

Comment (xla) [reply proxy/coco/src/project.rs]: radicle-dev/radicle-link#335

File: proxy/api/src/http/project.rs [hunk 4]
```diff
@@ -308,13 +304,30 @@ mod test {
...
+            .get_entry(coco::include::GIT_CONFIG_PATH_KEY)
+            .map_err(coco::Error::from)?
+            .value()
+            .unwrap()
+            .to_string();
+        assert_eq!(
+            include_path,
+            format!(
+                "{}/git-includes/{}.inc",
+                tmp_dir.path().display().to_string(),
```
Comment (FintanH) [inline proxy/api/src/http/project.rs:326]: Could make this more resilient using the `Paths` to fill in for `git-includes`

**Review by FintanH — APPROVED** [review]

**Review by rudolfs — APPROVED** [review]
Looks great!

A follow-up issue for some tests for this functionality would be nice. Perhaps could be just a note in https://github.com/radicle-dev/radicle-upstream/issues/908.

**Pull request merged**